src/argowrapper/engine/argo_engine.py

```diff
@@ -301,8 +303,12 @@ def _get_archived_workflow_given_name(self, archived_workflow_uid) -> str:
         workflow_details = self.get_workflow_details(None, archived_workflow_uid)
         #  get the workflow given name from the parsed details:
         given_name = workflow_details["wf_name"]
-        self.workflow_given_names_cache[archived_workflow_uid] = given_name
-        return given_name
+        team_project = workflow_details[GEN3_TEAM_PROJECT_METADATA_LABEL]
+        self.workflow_given_names_cache[archived_workflow_uid] = (
```

Comment:
For my information, is the purpose of this line to set the value in cache for subsequent reads?

Reply:
correct!